src/App.js

```diff
@@ -39,8 +39,8 @@ class App extends Component {
     return (
       <div className='main-page'>
         <Header />
-        <Route exact path='/:id' render={() =>
-          <MovieInfo id={this.state.cardID} changeDisplay={this.changeDisplay}/>
+        <Route exact path='/:id' render={({ match }) =>
```

Comment:
Good catch passing match as the render parameter so ID would be available to continue to use for API fetches